FirebaseAdmin/FirebaseAdmin/Auth/Providers/SamlProviderConfigArgs.cs

```diff
@@ -0,0 +1,142 @@
+        /// Gets or sets the SAML IdP entity identifier.
+        /// </summary>
+        public string IdpEntityId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the SAML IdP SSO URL.
+        /// </summary>
+        public string SsoUrl { get; set; }
+
+        /// <summary>
+        /// Gets or sets the collection of SAML IdP X.509 certificates issued by CA for this
+        /// provider. Multiple certificates are accepted to prevent outages during IdP key
+        /// rotation (for example ADFS rotates every 10 days). When the Auth server receives a SAML
+        /// response, it will match the SAML response with the certificate on record. Otherwise the
+        /// response is rejected. Developers are expected to manage the certificate updates as keys
+        /// are rotated.
+        /// </summary>
+        public IEnumerable<string> X509Certificates { get; set; }
+
+        /// <summary>
+        /// Gets or sets the SAML relying party (service provider) entity ID. This is defined by
+        /// the developer but needs to be provided to the SAML IdP.
+        /// </summary>
+        public string RpEntityId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the SAML callback URL. This is fixed and must always be the same as the
+        /// OAuth redirect URL provisioned by Firebase Auth,
+        /// <c>https://project-id.firebaseapp.com/__/auth/handler</c> unless a custom
+        /// <c>authDomain</c> is used. The callback URL should also be provided to the SAML IdP
+        /// during configuration.
+        /// </summary>
+        public string CallbackUrl { get; set; }
+
+        internal override AuthProviderConfig.Request ToCreateRequest()
+        {
+            var req = this.ToRequest();
+            if (string.IsNullOrEmpty(req.IdpConfig.IdpEntityId))
+            {
+                throw new ArgumentException("IDP entity ID must not be null or empty.");
+            }
+
+            if (string.IsNullOrEmpty(req.IdpConfig.SsoUrl))
+            {
+                throw new ArgumentException("SSO URL must not be null or empty.");
+            }
+            else if (!Uri.IsWellFormedUriString(req.IdpConfig.SsoUrl, UriKind.Absolute))
+            {
+                throw new ArgumentException($"Malformed SSO URL: {req.IdpConfig.SsoUrl}");
+            }
+
+            var certs = req.IdpConfig.IdpCertificates;
+            if (certs == null || certs.Count() == 0)
+            {
+                throw new ArgumentException("X509 certificates must not be null or empty.");
+            }
+            else if (certs.Any((cert) => string.IsNullOrEmpty(cert.X509Certificate)))
+            {
+                throw new ArgumentException(
+                    "X509 certificates must not contain null or empty values.");
+            }
+
+            if (string.IsNullOrEmpty(req.SpConfig.SpEntityId))
+            {
+                throw new ArgumentException("RP entity ID must not be null or empty.");
```

Comment:
Can I assume that RP/SP is interchangeable here?

Reply:
Internally everything is called `SP` (service provider) but in the public-facing API we use the `RP` (relying party) terminology.